sync/sync_head.go

```diff
@@ -136,33 +134,35 @@ func (s *Syncer[H]) setSubjectiveHead(ctx context.Context, netHead H) {
 
 // incomingNetworkHead processes new potential network headers.
 // If the header valid, sets as new subjective header.
-func (s *Syncer[H]) incomingNetworkHead(ctx context.Context, netHead H) pubsub.ValidationResult {
+func (s *Syncer[H]) incomingNetworkHead(ctx context.Context, netHead H) error {
 	// ensure there is no racing between network head candidates
 	s.incomingMu.Lock()
 	defer s.incomingMu.Unlock()
 	// first of all, check the validity of the netHead
-	res := s.validateHead(ctx, netHead)
-	if res == pubsub.ValidationAccept {
-		// and set it if valid
-		s.setSubjectiveHead(ctx, netHead)
+	err := s.validateHead(ctx, netHead)
+	if err != nil {
+		return err
 	}
-	return res
+	// and set it if valid
+	s.setSubjectiveHead(ctx, netHead)
+	return nil
 }
 
 // validateHead checks validity of the given header against the subjective head.
-func (s *Syncer[H]) validateHead(ctx context.Context, new H) pubsub.ValidationResult {
+func (s *Syncer[H]) validateHead(ctx context.Context, new H) error {
 	sbjHead, err := s.subjectiveHead(ctx)
 	if err != nil {
 		log.Errorw("getting subjective head during validation", "err", err)
-		return pubsub.ValidationIgnore // local error, so ignore
+		// local error, so uncertain
+		return &header.VerifyError{Reason: err, Uncertain: true}
 	}
-	// ignore header if it's from the past
 	if new.Height() <= sbjHead.Height() {
 		log.Warnw("received known network header",
 			"current_height", sbjHead.Height(),
 			"header_height", new.Height(),
 			"header_hash", new.Hash())
-		return pubsub.ValidationIgnore
+		// set uncertain, if it's from the past
+		return &header.VerifyError{Reason: err, Uncertain: true}
```

Comment:
shouldn't we be certain here, since it's from past? `uncertain: false` -- this means our subjective head is higher than the one received.